test/eventpattern.test.ts

```diff
@@ -0,0 +1,797 @@
+import { ebEventPatternTestCase, ebEventPatternTestCaseError } from "./util";
+import { EventBusEvent, EventPredicateFunction } from "../src/eventbridge";
+import { reflect } from "../src";
+
+type TestEvent = EventBusEvent<{
```

Comment:
We should encourage `interface`s instead of `type` aliases. It's good practice to use them since they statically store the type definition in the type system instead of lazily computing the definition. This produces better error messages and other introspection messages and is more efficient - something we need to be careful of.

Reply:
https://palantir.github.io/tslint/rules/interface-over-type-literal/

;-)

Reply:
And yeah, was going back and forth on how to generate the event structure and I didn't change this one after changing to a generic type.

Reply:
Actually, looks like interfaces cannot extend types with type arguments, so type it is.